src/lib/encoder.ts

```diff
@@ -403,10 +408,11 @@ export const valueContentEncodingMap = (valueContent: string) => {
       };
     }
     // NOTE: Deprecated. For reference/testing in future
+    // this valueContent does not exist
     case 'ArrayLength': {
       return {
-        type: 'uint256',
-        encode: (value: number | string) => padLeft(numberToHex(value), 64),
+        type: 'uint128',
+        encode: (value: number | string) => padLeft(numberToHex(value), 32),
         decode: (value: string) => hexToNumber(value),
```

Comment:
Then let's remove this.

It used to work at a time with uint256.
At no point it time it has or will exist with utin128 as it is already deprecated. I suggest to remove it totally.